symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigLoader.java

```diff
@@ -49,28 +57,45 @@ public static BdkConfig loadFromFile(String configPath) throws BdkConfigExceptio
    * @return Symphony Bot Configuration
    */
   public static BdkConfig loadFromInputStream(InputStream inputStream) throws BdkConfigException {
-    if (inputStream != null) {
-      JsonNode jsonNode = BdkConfigParser.parse(inputStream);
-      if (jsonNode != null) {
-        JSON_MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-        if (jsonNode.at("/botUsername").isMissingNode()) {
-          return JSON_MAPPER.convertValue(jsonNode, BdkConfig.class);
-        } else {
-          LegacySymConfig legacySymConfig = JSON_MAPPER.convertValue(jsonNode, LegacySymConfig.class);
-          return LegacyConfigMapper.map(legacySymConfig);
-        }
-      }
+    final BdkConfig bdkConfig = parseConfig(BdkConfigParser.parse(inputStream));
+    validateLoadBalancingConfiguration(bdkConfig);
+
+    return bdkConfig;
+  }
+
+  private static BdkConfig parseConfig(JsonNode jsonNode) {
+    if (jsonNode.at("/botUsername").isMissingNode()) {
+      return JSON_MAPPER.convertValue(jsonNode, BdkConfig.class);
+    } else {
+      LegacySymConfig legacySymConfig = JSON_MAPPER.convertValue(jsonNode, LegacySymConfig.class);
+      return LegacyConfigMapper.map(legacySymConfig);
+    }
+  }
+
+  private static void validateLoadBalancingConfiguration(BdkConfig config) throws BdkConfigException {
+    final BdkLoadBalancingConfig agentLoadBalancing = config.getAgentLoadBalancing();
+    if (agentLoadBalancing == null) {
+      return;
+    }
+
+    if (config.getAgent().overridesParentConfig()) {
+      throw new BdkConfigException("Both agent and lb-agent fields are defined");
```

Comment:
I guess this check will be bypassed if the lb config is setup programmatically, not from the yaml file. Then the `NullPointerException` will occur